### PR TITLE
 EPMCUXDA-6483: backend project structure: Prettier & Lint-staged & .editorconfig

### PR DESCRIPTION
**Pull request by mishelen**
As follows from headline it adds .editorconfig, which works as fallback for prettier and can improve formatting flow. But it cover some things uncovered by prettier: line-endings, charset...

Then prettier, previous setup only handles `js` files, all other types require to be overrided, also added `.prettierignore`.

After added `lint-staged` -- which allows to dont take care about formatting at all, when you commit all files will be formatted. It has 1 drawback - if you use partial commits, then you wont able to do that. 

Finally some files were prettified.

update: **seems that `lint-staged` already supports partial commits**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+[*]
+end_of_line = lf
+insert_final_newline = true
+# Set default charset
+charset = utf-8
+
+# 2 space indentation
+[*.{ts,json}]
+indent_style = space
+indent_size = 2
```
Comment (DenisDashkevich) [inline .editorconfig]: `trim_trailing_whitespace = true` ?

File: package.json [hunk 2]
```diff
@@ -7,6 +7,11 @@
     "build": "./node_modules/.bin/tsc",
     "lint": "./node_modules/.bin/eslint src/**/*.ts --fix"
   },
+  "lint-staged": {
+    "*.ts": ["eslint --fix", "git add"],
+    "*.{json,md}": ["prettier --write", "git add"],
+    "{.eslintrc}": ["prettier --write", "git add"]
+  },
```
Comment (DenisDashkevich) [inline package.json]: `.lintstagedrc` ?

File: package.json [hunk 3]
```diff
@@ -43,13 +48,14 @@
...
     "ts-node": "^8.0.3",
     "ts-node-dev": "^1.0.0-pre.32",
     "tslint": "^5.14.0",
     "typescript": "^3.3.4000"
-	},
-	"husky": {
+  },
+  "husky": {
     "hooks": {
       "pre-commit": "npm run lint"
```
Comment (DenisDashkevich) [inline package.json]: "precommit": "lint-staged"

Comment (DenisDashkevich) [reply package.json]: I suppose all hooks can be set just in `scripts`.

File: .editorconfig (hunk 1 above)
Comment (mishelen) [reply .editorconfig]: you, right. Wanted to add, but forgot. Sure, no problem

File: package.json (hunk 3 above)
Comment (mishelen) [reply package.json]: Regarding `lint-staged` you right, without it adding `lint-staged` miss all value. My bad, seems I forgot.

And about `hooks` placement, placing them in scripts will be deprecated soon.

File: package.json (hunk 2 above)
Comment (mishelen) [reply package.json]: Can be useful, do you have ideas what files needed to add? @DenisDashkevich

Comment (mishelen) [reply package.json]: oh, got you))

**Pull request merged**